src/commandSocket.cpp

```diff
@@ -91,11 +91,59 @@ class CommandSocket::Private
 
     std::unordered_map<int, std::shared_ptr<cura::proto::Layer>> sliced_layers;
 };
+
+class CommandSocket::PolygonCompiler
+{
+	static_assert(sizeof(PrintFeatureType) == 1, "To be compatible with the Cura frontend code PrintFeatureType needs to be of size 1");
+	CommandSocket::Private& _cs_private_data;
+	int _layer_nr;
+	std::vector<PrintFeatureType> _line_types;
+	std::vector<int> _line_widths;
+	cura::Polygon _poly;
+
+	PolygonCompiler(const PolygonCompiler&) = delete;
+	PolygonCompiler& operator=(const PolygonCompiler&) = delete;
+public:
+	PolygonCompiler(CommandSocket::Private& cs_private_data):
+		_cs_private_data(cs_private_data),
+		_layer_nr( 0 ),
+		_line_widths(),
+		_poly()
+	{}
+	~PolygonCompiler(){
+		if( _poly.size() )
+			flushPolygon();
```

Comment:
Please take note of our code conventions (https://github.com/Ultimaker/Meta/blob/master/code_conventions.md)
Most notable: now I will put your braces on new lines and introduce braces for one-line blocks of code which removes your credits from the git blame ;)